lib/features/workout/data/models/workout_model.dart

```diff
@@ -0,0 +1,200 @@
+class WorkoutModel with WorkoutModelMappable {
+  WorkoutModel({
+    required this.sections,
+    required this.id,
+    required this.title,
+    required this.subtitle,
+    required this.description,
+    required this.createdAt,
+    required this.updatedAt,
+  });
+
+  List<IWorkoutSection> sections;
+
+  final int id;
+  final String title;
+  final String subtitle;
+  final String description;
+
+  final DateTime updatedAt;
+  final DateTime createdAt;
+
+  WorkoutEntity toEntity() {
+    return WorkoutEntity(
+      title: title,
+      subtitle: subtitle,
+      description: description,
+      updatedAt: updatedAt,
+      createdAt: createdAt,
+    )..sections.addAll(sections.map((e) => e.toEntity()));
+  }
+}
+
+@MappableClass(discriminatorKey: 'organization')
+sealed class IWorkoutSection with IWorkoutSectionMappable {
+  WorkoutSectionEntity toEntity();
+}
+
+@MappableClass(discriminatorValue: 'default')
+class DefaultWorkoutSectionModel
+    with DefaultWorkoutSectionModelMappable
+    implements IWorkoutSection {
+  DefaultWorkoutSectionModel({
+    required this.id,
+    required this.title,
+    required this.sets,
+  });
+
+  final int id;
+  final String title;
+  List<IWorkoutSet> sets;
+
+  int get totalSets => sets.length;
+
+  int get minReps => sets.fold(99, (prev, e) {
+        switch (e) {
+          case final DefaultWorkoutSetModel defaultSet:
+            return min(prev, defaultSet.minReps);
+        }
+      });
+
+  int? get maxReps {
+    final maxReps = sets.fold(0, (prev, e) {
+      switch (e) {
+        case final DefaultWorkoutSetModel defaultSet:
+          return max(max(prev, defaultSet.minReps), defaultSet.maxReps ?? prev);
+      }
+    });
+
+    if (maxReps <= minReps) return null;
+    return maxReps;
+  }
+
+  @override
+  WorkoutSectionEntity toEntity() {
+    final defaultSection = DefaultSectionEntity(id: id, title: title)
+      ..sets.addAll(sets.map((e) => e.toEntity()));
+    return WorkoutSectionEntity(id: id)..defaultSection.target = defaultSection;
+  }
+}
+
+@MappableClass(discriminatorValue: 'superset')
+class SupersetWorkoutSectionModel
+    with SupersetWorkoutSectionModelMappable
+    implements IWorkoutSection {
+  SupersetWorkoutSectionModel({
+    required this.id,
+    required this.title,
+    required this.sets,
+  });
+
+  final int id;
+  final String title;
+  List<Map<String, IWorkoutSet>> sets;
+
+  Map<String, int> getTotalSets() {
+    Map<String, int> totalSets = {};
+
+    for (final set in sets) {
+      for (final entry in set.entries) {
+        if (totalSets.containsKey(entry.key)) {
+          totalSets[entry.key] = totalSets[entry.key]! + 1;
+          continue;
+        }
+
+        totalSets.addAll({entry.key: 1});
+      }
+    }
+
+    /// TODO: sorted and take first and last - maybe delete this later
+    Map<String, int> sortedTotalSets = SplayTreeMap.from(
+      totalSets,
+      (k1, k2) => totalSets[k1]!.compareTo(totalSets[k2]!),
+    );
+    return {
```

Comment:
What happens if all the exercises in the superset are done the same amount of times?  Would this return { A: 3, B: 3, C: 3 }

Reply:
added logic to return one total set if the first and last values are the same